### PR TITLE
[IA-4006] Add types for apps

### PR DESCRIPTION
**Pull request by jdcanas**
This PR does the following:
- adds types for Apps to conform to Leo API return
- Cleans up `tool-utils.ts`. The concept of a `tool` is used throughout the code base as a non-cloud-specific application with certain display implications. Really, all a `tool` is is a `label` that uniquely identifies it regardless of the underlying cloud implementation. The model in its current state approaches this implementation, but with a layer of indirection. Essentially, `app` related `tool`s had this nasty habit of having both a `label` and an `appType` that were exactly the same for properly-formed tools and not for inconsistent implementations. This is resolved and simplified by making the `label` for app related `tools` have a one-to-one mapping with the `appType` field returned from leonardo. This is a bit of a verbose description, so happy to chat further about this change since it affects multiple teams.
- Create test file for app-specific data transformations

An example of the code smell that arose prior to 2  is the tool `CromwellOnAzure`. This is not really a valid tool, `Cromwell` is the tool, and it can exist on multiple cloud platforms. I will have a follow-up PR to detangle this overloading, but this PR is noisy as is.

### CODE REVIEW

File: src/pages/workspaces/workspace/useDeleteWorkspaceState.ts [hunk 1]
```diff
@@ -11,25 +13,13 @@ import { isResourceDeletable } from 'src/pages/workspaces/workspace/analysis/uti
 
 export const WorkspaceResourceDeletionPollRate = 5000
 
-interface DeleteWorkspaceModalLeoApp {
```
Comment (aherbst-broad) [inline src/pages/workspaces/workspace/useDeleteWorkspaceState.ts:14]: Nice, thanks for cleaning this up.

**Review by aherbst-broad — APPROVED** [review]
Workspace deletion hook changes look good.

**Review by n8landolt — APPROVED** [review]

**Comment by salonishah11** [comment]
@jdcanas It seems the `Settings` option is now being displayed for Cromwell on Azure apps. It needs to stay hidden. Maybe the `isSettingsUnsupported` is not being correctly checked as expected? Also the tooltip for `Open` has Cromwell in uppercase letters. Can they be made camel case?
Cromwell on Azure modal from dev branch:
Screen Shot 2023-03-06 at 5 49 46 PM

Cromwell on Azure modal from this branch:
Screen Shot 2023-03-06 at 5 57 14 PM

**Comment by jdcanas** [comment]
> @jdcanas It seems the `Settings` option is now being displayed for Cromwell on Azure apps. It needs to stay hidden. Maybe the `isSettingsUnsupported` is not being correctly checked as expected? Also the tooltip for `Open` has Cromwell in uppercase letters. Can they be made camel case? Cromwell on Azure modal from dev branch: Screen Shot 2023-03-06 at 5 49 46 PM
> 
> Cromwell on Azure modal from this branch: Screen Shot 2023-03-06 at 5 57 14 PM

Try now @salonishah11? Not sure how it happened, but I refactored to make the logic a bit more robust.

**Comment by salonishah11** [comment]
@jdcanas It seems that now the `Settings` option has been hidden for all apps on both Google and Azure workspaces
Modal in Google workspace:
Screen Shot 2023-03-08 at 2 16 29 PM

Modal in Azure workspace:
Screen Shot 2023-03-08 at 2 17 06 PM

It also seems that clicking on the `Open` button in Cromwell app no longer opens the CBAS UI in new tab even though the app is running fine. Clicking on `Open` just dismisses the modal. Manually pasting the cbas-ui proxy url (from network tab) in new tab does load the CBAS UI. Can you verify if you are seeing this behavior too?

**Review by n8landolt — APPROVED** [review]

**Review by salonishah11 — APPROVED** [review]
Workflows/Cromwell related changes LGTM. Thank you for refactoring Cromwell apps!

**Pull request merged**